### PR TITLE
[Intl] Add support for ISO 3166-1 Alpha-3 country codes

### LINKED ISSUE #20313: [Intl][Form] Support ISO 3166-1 Alpha-3 country codes
Could we please extend the form type `CountryType` via an option to make it support ISO_3166-1 Alpha-3 country codes?

Sources for the country codes:
- https://www.iso.org/obp/ui/
- http://en.wikipedia.org/wiki/ISO_3166-1_alpha-3
- https://github.com/thephpleague/iso3166
- https://gist.github.com/brianherbert/1687655

I'm currently working on a project where I need country codes as 3 characters instead of just 2.

Would it be possible to include a new alpha3 option or something to support 3 characters country codes? 

(This was first raised in #8492 but that issue was closed without being resolved properly)

### PR DESCRIPTION
**Pull request by TerjeBr**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes 
| Fixed tickets | #20313
| License       | MIT

This PR is just to change the INTL component.

More work will be needed in the Form component to fully fix #20313

### CODE REVIEW

**Comment by stof** [comment]
you have some merge commit that should be be there in your branch.

File: src/Symfony/Component/Intl/Languages.php [hunk 1]
```diff
@@ -75,6 +75,11 @@ public static function getAlpha3Code(string $language): string
         return self::readEntry(['Alpha2ToAlpha3', $language], 'meta');
     }
 
+    public static function getAlpha3Codes(): array
```
Comment (stof) [inline src/Symfony/Component/Intl/Languages.php]: please add the phpdoc. We need at least `@return string[]` to document the return type more precisely.

And we need tests covering this API

File: src/Symfony/Component/Intl/Countries.php [hunk 2]
```diff
@@ -65,6 +104,20 @@ public static function getNames($displayLocale = null)
         return self::asort(self::readEntry(['Names'], $displayLocale), $displayLocale);
     }
 
+    /**
+     * @return string[]
```
Comment (stof) [inline src/Symfony/Component/Intl/Countries.php]: this needs more documentation to document what it returns. This returns an array of names indexed by alpha3 code, which is not obvious from the method name.

**Comment by terjebraten-certua** [comment]
> 
> 
> you have some merge commit that should be be there in your branch.

I guess you mean it should not be there. I had to merge in the 4.4 branch from the upstream repo. I do not know how I can remove this merge commit from the PR. Have you got any pointes?

**Comment by stof** [comment]
you should have done a fast-forward merge, not a merge commit.

**Comment by stof** [comment]
See https://symfony.com/doc/current/contributing/code/pull_requests.html#rebase-your-pull-request for the doc about how to rebase your PR to clean it from this merge commit.

**Comment by terjebraten-certua** [comment]
The problem was that my fork of symfony did not have a 4.4 branch.

How can I fix this then? Do I need to delete my current fork and make a new fork?

File: src/Symfony/Component/Intl/Countries.php (hunk 2 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Countries.php]: done

**Comment by Simperfit** [comment]
You just need to create new branch from symfony:4.4 (adding the  symfony/symfony into a symfony remote for exemple).

**Comment by ro0NL** [comment]
Note im not yet convinced we need anything else then `getAlpha3Code(string $alpha2Code)` (both in Languages and Countries). That's enough for form integration.

Do we have a usecase for the other methods?

From #32663; please compile the dataset (see the `compile` binary). IMHO feel free to investigate the language metadata, to replace the custom mapping in core.

**Comment by terjebraten-certua** [comment]
> 
> 
> You just need to create new branch from symfony:4.4 (adding the symfony/symfony into a symfony remote for exemple).

I cannot create any new branch from 4.4 unless I have a fork of symfony on ghithub with the 4.4 branch in it. The problem was that I had an old fork of symfony on github without the 4.4 branch. And github will not let me create another fork. I do not know how to tell github to update my fork. If I could only tell github to pull from the upstream repo, but I see no button to that in the web interface.

**Comment by stof** [comment]
@terjebraten-certua you don't need a 4.4 branch in your github fork. You need it in your local clone of Symfony.

And the way to update the branch using fast-forward is to do it locally, and then push the updated branch to your github fork. there is no github UI to do it.

**Comment by Simperfit** [comment]
```
git remote add symfony git@github.com:symfony/symfony.git 
git fetch --alll
git checkout symfony/4.4
git checkout -b the-new-brand-from-4.4
```

**Comment by terjebraten-certua** [comment]
```
tbraten@SGDSK03 MINGW64 /d/Source/symfony (4.4)
$ git remote add symfony git@github.com:symfony/symfony.git

tbraten@SGDSK03 MINGW64 /d/Source/symfony (4.4)
$ git fetch --all
Fetching origin
FATAL ERROR: Couldn't agree a key exchange algorithm (available: curve25519-sha2                          56,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-                          nistp521)
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
error: Could not fetch origin
Fetching symfony
FATAL ERROR: Couldn't agree a key exchange algorithm (available: curve25519-sha2                          56,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-                          nistp521)
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
error: Could not fetch symfony

tbraten@SGDSK03 MINGW64 /d/Source/symfony (4.4)
```

**Comment by terjebraten-certua** [comment]
> Note im not yet convinced we need anything else then `getAlpha3Code(string $alpha2Code)` (both in Languages and Countries). That's enough for form integration.

Yes, that is strictly speaking enough. But why this assymetry? Should not alpha3 have the same support fuctions as alpha2?

> 
> Do we have a usecase for the other methods?
The usecase is that it makes it easier to work with alpha3 without relying on alpha2.

If you need a list of all alpha3 codes, it is a bit inefficent to first get the list of alpha2 codes and then convert them one by one to alpha3.

> 
> From #32663; please compile the dataset (see the `compile` binary).
I did that.
The only difference is in the meta.json that is in this diff.

> IMHO feel free to investigate the language metadata as well, to replace the custom mapping in core.
Which custom mapping are you talking about?

File: src/Symfony/Component/Intl/Languages.php (hunk 1 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Languages.php]: Done

**Comment by ro0NL** [comment]
> > Note im not yet convinced we need anything else then `getAlpha3Code(string $alpha2Code)` (both in Languages and Countries). That's enough for form integration.
> 
> Yes, that is strictly speaking enough. But why this assymetry? Should not alpha3 have the same support fuctions as alpha2?

Maybe, ive no real opinion :) in any case we should prefer the same API between languages and countries. 

> > Do we have a usecase for the other methods?
> 
> The usecase is that it makes it easier to work with alpha3 without relying on alpha2.
> 
> If you need a list of all alpha3 codes, it is a bit inefficent to first get the list of alpha2 codes and then convert them one by one to alpha3.

Fair enough, i tend to believe working with alpha2 codes is the common case, but i could be wrong here.

> > From #32663; please compile the dataset (see the `compile` binary).
> 
> I did that.
> The only difference is in the meta.json that is in this diff.

Sorry missed it :+1: 

> > IMHO feel free to investigate the language metadata as well, to replace the custom mapping in core.
> 
> Which custom mapping are you talking about?

See https://github.com/symfony/symfony/blob/a7852c0da89d926889c430d57e91f740dd6f1b62/src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php#L32

File: src/Symfony/Component/Intl/Countries.php [hunk 3]
```diff
@@ -58,13 +98,44 @@ public static function getName(string $country, string $displayLocale = null): s
     }
 
     /**
+     * Get country name from alpha3 code.
+     *
+     * @throws MissingResourceException if the country code does not exists
+     */
+    public static function getNameFromAlpha3(string $country, string $displayLocale = null): string
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: `getAlpha3Name`

File: src/Symfony/Component/Intl/Countries.php [hunk 4]
```diff
@@ -58,13 +98,44 @@ public static function getName(string $country, string $displayLocale = null): s
...
     }
 
+    /**
+     * Get list of country names indexed with alpha3 codes as keys.
+     *
+     * Same as method getNames, but with alpha3 codes instead of alpha2 codes as keys.
+     *
+     * @return string[]
+     */
+    public static function getAlpha3AndNames($displayLocale = null): array
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: `getAlpha3Names`

File: src/Symfony/Component/Intl/Countries.php [hunk 5]
```diff
@@ -58,13 +98,44 @@ public static function getName(string $country, string $displayLocale = null): s
...
+    {
+        $alpha2Names = self::readEntry(['Names'], $displayLocale);
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: i'd call `getNames` and map the keys, no need to asort here

**Review by ro0NL — COMMENTED** [review]
note CI doesnt run intl-data tests, you need to verify locally with `--group intl-data`

File: src/Symfony/Component/Intl/Countries.php [hunk 6]
```diff
@@ -50,6 +73,23 @@ public static function exists(string $country): bool
     }
 
     /**
+     * @param string $country Alpha3 country code
+     * @return bool
+     */
+    public static function alpha3exists(string $country): bool
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: `alpha3CodeExists`?

**Comment by terjebraten-certua** [comment]
> in any case we should prefer the same API between languages and countries.
Should I add those same methods to the Languages class then?


> 
> See
> 
> https://github.com/symfony/symfony/blob/a7852c0da89d926889c430d57e91f740dd6f1b62/src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php#L32

That is list of preferred alpha3 codes when there is more than one code to choose from.

See 

https://github.com/symfony/symfony/blob/a7852c0da89d926889c430d57e91f740dd6f1b62/src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php#L201

**Comment by ro0NL** [comment]
> Should I add those same methods to the Languages class then?

IMHO yes, it targets the same domain logic

> That is list of preferred alpha3 codes when there is more than one code to choose from.

Ouch :sweat: all good then.

**Comment by terjebraten-certua** [comment]
I do not have more time to work on this now.

Anyone feel free to add tests to my PR and complete it.

May be I will have some more time to finish this in the coming week end.

File: src/Symfony/Component/Intl/Countries.php [hunk 7]
```diff
@@ -58,13 +100,46 @@ public static function getName(string $country, string $displayLocale = null): s
     }
 
     /**
+     * Get country name from alpha3 code.
+     *
+     * @throws MissingResourceException if the country code does not exists
+     */
+    public static function getAlpha3Name(string $country, string $displayLocale = null): string
+    {
+        $alpha2Code = self::readEntry(['Alpha2ToAlpha3', $country], 'meta');
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: given we pass an alpha3 code, we need a lookup by value to get us the corresponding key (which is the alpha2 code). This wont work IIUC.

Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Countries.php]: Yes, I have changed that.

**Comment by terjebraten-certua** [comment]
This PR should now be ready to merge.

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php [hunk 8]
```diff
@@ -125,14 +137,25 @@ protected function generateDataForRoot(BundleEntryReaderInterface $reader, $temp
...
 
+        $alpha2ToAlpha3 = $this->generateAlpha3($metadataBundle);
+
         sort($this->regionCodes);
+        $unIndexed = $this->regionCodes;
+
+        $this->regionCodes = [];
+        foreach ($unIndexed as $alpha2Code) {
+            $alpha3code = $alpha2ToAlpha3[$alpha2Code];
+            $this->regionCodes[$alpha3code] = $alpha2Code;
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: this alters the keys in `Countries::getCountryCodes()` (im not sure we should do it), whereas for languages this is compiled separate

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php [hunk 9]
```diff
@@ -162,11 +162,14 @@ protected function generateDataForMeta(BundleEntryReaderInterface $reader, $temp
...
 
+        $alpha2ToAlpha3 = $this->generateAlpha2ToAlpha3Mapping($metadataBundle);
+
         return [
             'Version' => $rootBundle['Version'],
             'Languages' => $this->languageCodes,
             'Aliases' => array_column(iterator_to_array($metadataBundle['alias']['language']), 'replacement'),
-            'Alpha2ToAlpha3' => $this->generateAlpha2ToAlpha3Mapping($metadataBundle),
+            'Alpha2ToAlpha3' => $alpha2ToAlpha3,
+            'Alpha3ToAlpha2' => array_flip($alpha2ToAlpha3),
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: i think we have to decide whether to compile both mappings, compared to a single source of truth and cache the reverse map in-memory. I tend to prefer the latter and keep the resource files small as possible (unless it's a significant perf gain)

**Review by ro0NL — COMMENTED** [review]
(due the re-ordering it may need a re-compile for upper branches)

File: src/Symfony/Component/Intl/Countries.php [hunk 10]
```diff
@@ -31,13 +31,42 @@ final class Countries extends ResourceBundle
...
+
+    public static function getAlpha2Code(string $alpha3Code)
+    {
+        return self::readEntry(['Regions', $alpha3Code], 'meta');
+    }
+
+    /**
+     * @param string $country Alpha2 country code
+     *
+     * @return bool
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: all these can be removed

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php [hunk 11]
```diff
@@ -204,6 +207,7 @@ private function generateAlpha2ToAlpha3Mapping(ArrayAccessibleResourceBundle $me
                 }
             }
         }
+        asort($alpha2ToAlpha3);
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: actually maybe this should be done in 3.4 even, to avoid future conflicts

**Review by ro0NL — COMMENTED** [review]
some phpdoc fixes can be applied for 4.3 as well (not sure we really care)

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php (hunk 9 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: So, flipping an array of 250 languages may not take long if you only do it once. But it might be a pain if it is inside some loop in the application code.

This also made the implementation much easier.

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php (hunk 8 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: I thought it would be useful to have the 3 letter codes as keys.

Comment (ro0NL) [reply src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: we ask for a list of alpha2 codes (`getCountryCodes`), i wouldnt mix&match both data sets

**Comment by terjebraten-certua** [comment]
> (due the re-ordering it may need a re-compile for upper branches)

I did not understand that. What upper branches are you talking about?

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php (hunk 9 above)
Comment (ro0NL) [reply src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: but we do it once :) e.g. `private Countries::getAlpha3ToAlpha2()` which caches the flipped result in a static variable/property

**Comment by ro0NL** [comment]
right, i think we should re-order in 3.4 first and merge it towards 4.4. We can also keep the current ordering to avoid the hassle

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php (hunk 8 above)
Comment (TerjeBr) [reply src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: Well, I was thinking about the symmetry with the `getAlpha3Codes` method where the alpha2 codes are keys.

It also makes it easier to implement `getAplha3Name` and `alpha3CodeExists`.

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php (hunk 9 above)
Comment (TerjeBr) [reply src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: This readEntry methods seems to be advanced classes that do caching for you. If we want to take advantage of that I think we need two mappings on disk. It is not only the data, but also the alphabetic ordering. Does not that justify 2 datasets? I think it does.

And it is generated data anyway, the one source of truth is in the ISO bundle.

**Comment by TerjeBr** [comment]
The current ordering is not alphabetic. I think that is worth having.

**Comment by TerjeBr** [comment]
@ro0NL I think I am satisified with what I have have done. But you seem to have some other ideas on how the data should be organized. May be the best thing will be if you open a separate PR that build on mine. Then you can organize it the way you want it, and I do not have to guess how it should be.

**Comment by ro0NL** [comment]
> Then you can organize it the way you want it, and I do not have to guess how it should be.

im inspired by your work :) and trying to find the right fit.

Agree we can compile the mappings in both directions, and keep the implementation simple :+1: (it's a single meta file anyway)

Im still skeptical about https://github.com/symfony/symfony/pull/32676#discussion_r310197278 though, i see no reason to have different behavior between Countries and Languages, nor the behavior change in getCountryCodes() regarding the keys.

To be on the safe/explicit side i suggest to consistently compile `Alpha2ToAlpha3/Alpha3ToAlpha2` for both languages and countries, and keep `Languages/Regions` as is

**Comment by terjebraten-certua** [comment]
Ok, have now changed the code according to your latest comment.

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php (hunk 8 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: The returnvalue from `getCountryCodes` is now the same as before.

**Comment by terjebraten-certua** [comment]
And thank you @ro0NL for taking care of the reordering in the parent branches.

**Comment by ro0NL** [comment]
@terjebraten-certua can you rebase please?

**Comment by terjebraten-certua** [comment]
Ok. I have rebased and squashed my commits.

File: src/Symfony/Component/Intl/Countries.php [hunk 12]
```diff
@@ -38,6 +38,30 @@ public static function getCountryCodes(): array
...
+    {
+        return self::readEntry(['Alpha2ToAlpha3'], 'meta');
+    }
+
+    public static function getAlpha3Code(string $alpha2Code): string
+    {
+        return self::readEntry(['Alpha2ToAlpha3', $alpha2Code], 'meta');
+    }
+
+    public static function getAlpha2Code(string $alpha3Code)
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: `:string`

File: src/Symfony/Component/Intl/Countries.php [hunk 13]
```diff
@@ -62,6 +100,18 @@ public static function getName(string $country, string $displayLocale = null): s
...
+    /**
+     * Get country name from alpha3 code.
+     *
+     * @throws MissingResourceException if the country code does not exists
+     */
+    public static function getAlpha3Name(string $alpha3Code, string $displayLocale = null): string
+    {
+        $alpha2Code = self::getAlpha2Code($alpha3Code);
+
+        return self::readEntry(['Names', $alpha2Code], $displayLocale);
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: `return self::getName(self::getAlpha2Code($alpha3Code))` without `@throws` here

File: src/Symfony/Component/Intl/Countries.php [hunk 14]
```diff
@@ -72,6 +122,25 @@ public static function getNames($displayLocale = null): array
...
+     *
+     * @return string[]
+     */
+    public static function getAlpha3Names($displayLocale = null): array
+    {
+        $alpha2Names = self::getNames($displayLocale);
+        $alpha3Names = [];
+        foreach ($alpha2Names as $alpha2Code => $name) {
+            $alpha3Code = self::readEntry(['Alpha2ToAlpha3', $alpha2Code], 'meta');
+            $alpha3Names[$alpha3Code] = $name;
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Countries.php]: `$alpha3Names[self::getAlpha3Code($alpha2Code)] = $name`

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php [hunk 15]
```diff
@@ -204,7 +209,6 @@ private function generateAlpha2ToAlpha3Mapping(ArrayAccessibleResourceBundle $me
                 }
             }
         }
-
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: unrelated

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php [hunk 16]
```diff
@@ -125,14 +137,25 @@ protected function generateDataForRoot(BundleEntryReaderInterface $reader, $temp
...
         $rootBundle = $reader->read($tempDir, 'root');
+        $metadataBundle = $reader->read($tempDir, 'metadata');
 
         $this->regionCodes = array_unique($this->regionCodes);
 
+        $alpha2ToAlpha3 = $this->generateAlpha3($metadataBundle);
+
         sort($this->regionCodes);
 
+        $alpha3ToAlpha2 = [];
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: cant we do `$alpha3ToAlpha2 = array_flip($alpha2ToAlpha3)` like in languages?

File: src/Symfony/Component/Intl/Languages.php [hunk 17]
```diff
@@ -79,6 +79,81 @@ public static function getAlpha3Code(string $language): string
...
+    }
+
+    /**
+     * Get language name from ISO 639-2 three-letter code.
+     *
+     * @throws MissingResourceException if the country code does not exists
+     */
+    public static function getAlpha3Name(string $language, string $displayLocale = null): string
+    {
+        $alpha2Code = self::readEntry(['Alpha3ToAlpha2', $language], 'meta');
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Languages.php]: `return self::getName(self::getAlpha2Code($language))` without `@throws` here

File: src/Symfony/Component/Intl/Languages.php [hunk 18]
```diff
@@ -79,6 +79,81 @@ public static function getAlpha3Code(string $language): string
...
+        return self::readEntry(['Alpha2ToAlpha3'], 'meta');
+    }
+
+    /**
+     * @param string $language ISO 639-2 three-letter language code
+     */
+    public static function alpha3CodeExists(string $language): bool
+    {
+        try {
+            self::readEntry(['Alpha3ToAlpha2', $language], 'meta');
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Languages.php]: `self::getAlpha2Code($language)`

**Comment by terjebraten-certua** [comment]
Squashed the merge fix and code standard fix.

File: src/Symfony/Component/Intl/Languages.php [hunk 19]
```diff
@@ -79,6 +79,81 @@ public static function getAlpha3Code(string $language): string
...
+     * Same as method getNames, but with ISO 639-2 three-letter codes instead of ISO 639-1 codes as keys.
+     *
+     * @return string[]
+     */
+    public static function getAlpha3Names($displayLocale = null): array
+    {
+        $alpha2Names = self::getNames($displayLocale);
+        $alpha3Names = [];
+        foreach ($alpha2Names as $alpha2Code => $name) {
+            if (2 === \strlen($alpha2Code)) {
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Languages.php]:
```php
try {
    $alpha3Names[self::getAlpha3Code($alpha2Code)] = $name;
} catch (MissingResourceException $e) {
}
```

**Review by ro0NL — COMMENTED** [review]
last round from me. Not sure what happened during review, but please revert the faulty commits :)

**Comment by terjebraten-certua** [comment]
Sorry. somhow my code got rebased on wrong branch

File: src/Symfony/Component/Intl/Countries.php (hunk 13 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Countries.php]: How can it be without `@throws` ?

Comment (ro0NL) [reply src/Symfony/Component/Intl/Countries.php]: AFAIK we dont document indirect throws

Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Countries.php]: That is news to me. Thought a method had to document all throws, no matter where they come from.

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php (hunk 16 above)
Comment (terjebraten-certua) [reply src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: I just wanted to be 100% sure that we had all the entries that was in `$this->regionCodes`.

Since this is a generator I want to code to be accurate instead of efficient.

**Comment by terjebraten-certua** [comment]
Closing in favour of https://github.com/symfony/symfony/pull/32988

Sorry, I could not rescue my branch that had become rebased somehow on master instead fo 4.4.
